liquibase-integration-tests/src/test/groovy/liquibase/command/IncludePreconditionsTest.groovy

```diff
@@ -0,0 +1,68 @@
+package liquibase.command
+
+import liquibase.Scope
+import liquibase.changelog.ChangeLogParameters
+import liquibase.command.core.UpdateCommandStep
+import liquibase.command.core.helpers.DatabaseChangelogCommandStep
+import liquibase.command.core.helpers.DbUrlConnectionArgumentsCommandStep
+import liquibase.command.util.CommandUtil
+import liquibase.extension.testing.testsystem.DatabaseTestSystem
+import liquibase.extension.testing.testsystem.TestSystemFactory
+import liquibase.extension.testing.testsystem.spock.LiquibaseIntegrationTest
+import liquibase.resource.SearchPathResourceAccessor
+import liquibase.ui.ConsoleUIService
+import spock.lang.Shared
+import spock.lang.Specification
+
+/**
+ * Include preconditions test
+ * @author Edoardo Patti
+ */
+@LiquibaseIntegrationTest
+class IncludePreconditionsTest extends Specification {
+
+    @Shared
+    private DatabaseTestSystem h2 = Scope.currentScope.getSingleton(TestSystemFactory).getTestSystem("h2") as DatabaseTestSystem
+
+    def "run include with preconditions"() {
+        given:
+        ConsoleUIService console = Scope.getCurrentScope().getUI() as ConsoleUIService
+        def outputStream = new ByteArrayOutputStream()
+        console.setOutputStream(new PrintStream(outputStream))
+        when:
+        String changelogFile = "changelogs/h2/include/master.xml"
+        def changelog =
+                DatabaseChangelogCommandStep.getDatabaseChangeLog(changelogFile, new ChangeLogParameters(), h2.getDatabaseFromFactory())
+        def resourceAccessor = new SearchPathResourceAccessor("target/test-classes")
+        def scopeSettings = [
+                (Scope.Attr.resourceAccessor.name()) : resourceAccessor
+        ]
+
+        Scope.child(scopeSettings, {
+            def updateCommand = new CommandScope(UpdateCommandStep.COMMAND_NAME)
+            updateCommand.addArgumentValue(DbUrlConnectionArgumentsCommandStep.DATABASE_ARG, h2.getDatabaseFromFactory())
+            updateCommand.addArgumentValue(UpdateCommandStep.CHANGELOG_ARG, changelog)
+            updateCommand.execute()
+        } as Scope.ScopedRunner)
+        String outputString = outputStream.toString()
+        def changelogResultSet = h2.getConnection().createStatement().executeQuery("select * from databasechangelog")
+        h2.getConnection().createStatement().executeQuery("select * from INCLUDED_TABLE")
+        h2.getConnection().createStatement().executeQuery("select * from DEFAULT_TABLE")
+        def executed = Arrays.stream(outputString.split("\r\n")).filter {
```

Comment:
Just an observation here: this regex doesn't work correctly for me in macOS. For me, this line is not filtering correctly and making the test fail, only using '\n' does it.

Reply:
Yes, that's because "Microsoft loves linux"...but not so much 😄
resolved with c433c0a42fca80ea39090cb4224625d9eb5dfa98